arrows/vxl/image_io.cxx

```diff
@@ -138,6 +142,97 @@ convert_image_helper(const vil_image_view<bool>& src,
   dest = src;
 }
 
+
+std::string
+plane_filename(std::string filename, unsigned p)
+{
+  std::string parent_directory =
+    kwiversys::SystemTools::GetParentDirectory(filename);
+  std::string file_name_with_ext =
+    kwiversys::SystemTools::GetFilenameName(filename);
+
+  std::size_t last_index = file_name_with_ext.find_last_of( "." );
```

Comment:
use `GetFilenameLastExtension()` and `GetFilenameWithoutLastExtension()` instead